src/strands/ir_builders.js

```diff
@@ -463,40 +464,72 @@ export function swizzleTrap(id, dimension, strandsContext) {
-            strandsContext,
-            {baseType: originalNode.baseType, dimension: null}, 
-            value
-          ).mappedDependencies;
-          const newDeps = new Array(originalNode.dimension);
-          for (let i = 0; i < target.dimension; i++) {
-            console.log(changed[i])
-            if (changed[i] !== undefined) {
-              newDeps[i] = newValues[changed[i]];
-            }
+  set(target, property, value, receiver) {
+    for (const basis of swizzleSets) {
+      const chars = [...property];
+      const valid =
+        chars.every(c => basis.includes(c)) &&
+        new Set(chars).size === chars.length &&
+        target.dimension >= chars.length;
+
+      if (!valid) continue;
+
+      const dim = target.dimension;
+
+      const lanes = new Array(dim);
+      for (let i = 0; i < dim; i++) {
+        const { id, dimension } = swizzleNode(strandsContext, target, 'xyzw'[i]);
+        lanes[i] = createStrandsNode(id, dimension, strandsContext);
+      }
+
+      let scalars = [];
+      if (value instanceof StrandsNode) {
+        if (value.dimension === 1) {
+          scalars = Array(chars.length).fill(value);
+        } else if (value.dimension >= chars.length) {
+          for (let k = 0; k < chars.length; k++) {
+            const { id, dimension } = swizzleNode(strandsContext, value, 'xyzw'[k]);
+            scalars.push(createStrandsNode(id, dimension, strandsContext));
           }
-          console.log("AFTER: ", newDeps)
-          // primitiveConstructorNode(strandsContext, {baseType, dimension}, newDeps);
-          return true;
+        } else {
+          FES.userError('type error', `Swizzle assignment: RHS vector too short (need ${chars.length}, got ${value.dimension}).`);
+        }
+      } else if (Array.isArray(value)) {
+        const flat = value.flat(Infinity);
+        if (flat.length === 1) {
+          scalars = Array(chars.length).fill(flat[0]);
+        } else if (flat.length === chars.length) {
+          scalars = flat;
+        } else {
+          FES.userError('type error', `Swizzle assignment: RHS length ${flat.length} does not match ${chars.length}.`);
         }
+      } else if (typeof value === 'number') {
+        scalars = Array(chars.length).fill(value);
+      } else {
+        FES.userError('type error', `Unsupported RHS for swizzle assignment: ${value}`);
       }
-      return Reflect.set(...arguments);
+
+      for (let j = 0; j < chars.length; j++) {
+        const canonicalIndex = basis.indexOf(chars[j]);
+        lanes[canonicalIndex] = scalars[j];
+      }
+
+      const orig = DAG.getNodeDataFromID(strandsContext.dag, target.id);
+      const baseType = orig?.baseType ?? BaseType.FLOAT;
+      const { id: newID } = primitiveConstructorNode(
+        strandsContext,
+        { baseType, dimension: dim },
+        lanes
+      );
+
+      target.id = newID;
+      if (typeof onRebind === 'function') {
+        onRebind(newID);
```

Comment:
This is the bit that ensures dependency ordering works after reassigning a property right? Maybe comment around here explaining what this is for